abipy/lessons/lesson_helper_functions.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+"""
+reusable functions for lessons
+"""
+
+from __future__ import division, print_function
+
+import sys
+import os
+import shutil
+import abipy.abilab as abilab
+import abipy.data as abidata
+
+
+def help(stream=sys.stdout):
+    """
+    Display the tutorial text.
+    """
+    stream.write(__doc__)
```

Comment:
code duplication, and there are by now already 3 of them, most likely there will be more later on ..

Michiel van Setten
UCL, NAPS
mjvansetten@gmail.com
michiel.vansetten@uclouvain.be
010 478824

'"Although the way may not be obvious at first unless you're Dutch"
from The Zen of Python

On 21 Jan 2015, at 14:54, gmatteo wrote:

> In abipy/lessons/lesson_helper_functions.py:
> 
> > +"""
> > +
> > +from __future__ import division, print_function
> > +
> > +import sys
> > +import os
> > +import shutil
> > +import abipy.abilab as abilab
> > +import abipy.data as abidata
> > +
> > +
> > +def help(stream=sys.stdout):
> > +    """
> > +    Display the tutorial text.
> > +    """
> > +    stream.write(__doc__)
> help access a global variable hence the function must be defined in the module in which this global variable is defined. For sure there are other tricks (introspection) that can be used to solve the problem but why? help is a very small function so I don't see any problem if we put these three lines of code in the different module
> …
> —
> Reply to this email directly or view it on GitHub.
>